src/frontend/src/optimizer/plan_node/logical_union.rs

```diff
@@ -13,7 +13,8 @@
 // limitations under the License.
 
 use std::cmp::max;
-use std::collections::BTreeMap;
+use std::collections::HashMap;
```

Comment:
Maybe the planner test failure was due to indeterminism of `HashMap`. I will keep investigating.